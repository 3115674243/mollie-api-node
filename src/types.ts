import createMollieClient from './createMollieClient';
export default createMollieClient;

export * from './createMollieClient';

export { default as List } from './data/list/List';

export { default as Capture } from './data/payments/captures/Capture';
import {
  GetParameters as CapturesGetParameters,
  ListParameters as CapturesListParameters
} from './resources/payments/captures/parameters';
export { CapturesGetParameters, CapturesListParameters };

export { default as Chargeback } from './data/chargebacks/Chargeback';
import {
  ListParameters as ChargebacksListParameters
} from './resources/chargebacks/parameters';
export { ChargebacksListParameters };

export { default as Customer } from './data/customers/Customer';
import {
  CreateParameters as CustomerCreateParams,
  GetParameters as CustomerGetParams,
  ListParameters as CustomersListParams,
  UpdateParameters as CustomerUpdateParams,
  DeleteParameters as CustomerDeleteParams,
} from './resources/customers/parameters';
export { CustomerCreateParams, CustomerGetParams, CustomersListParams, CustomerUpdateParams, CustomerDeleteParams };

export { default as Mandate } from './data/customers/mandates/Mandate';
import { CreateParameters as MandateCreateParams, GetParameters as MandateGetParams, ListParameters as MandatesListParams, RevokeParameters as MandateRevokeParams } from './resources/customers/mandates/parameters';
export { MandateCreateParams, MandateGetParams, MandatesListParams, MandateRevokeParams };

export { default as Method } from './data/methods/Method';
import { GetParameters as MethodGetParams, ListParameters as MethodsListParams } from './resources/methods/parameters';
export { MethodGetParams, MethodsListParams };

export { default as Order } from './data/orders/Order';
import {
  CreateParameters as OrderCreateParams,
  GetParameters as OrderGetParams,
  ListParameters as OrdersListParams,
  UpdateParameters as OrderUpdateParams,
  CancelParameters as OrderCancelParams,
} from './resources/orders/parameters';
export { OrderCreateParams, OrderGetParams, OrdersListParams, OrderUpdateParams, OrderCancelParams };

export { default as OrderLine } from './data/orders/orderlines/OrderLine';
import { UpdateParameters as OrderLineUpdateParams, CancelParameters as OrderLineCancelParams } from './resources/orders/orderlines/parameters';
export { OrderLineUpdateParams, OrderLineCancelParams };

export { default as Payment } from './data/payments/Payment';
import { CreateParameters as PaymentCreateParams, GetParameters as PaymentGetParams, ListParameters as PaymentsListParams, CancelParameters as PaymentCancelParams } from './resources/payments/parameters';
import { CreateParameters as CustomerPaymentCreateParams, ListParameters as CustomerPaymentsListParams } from './resources/customers/payments/parameters';
import { CreateParameters as OrderPaymentCreateParams } from './resources/payments/orders/parameters';
export {
  PaymentCreateParams,
  PaymentGetParams,
  PaymentsListParams,
  PaymentCancelParams,
  CustomerPaymentCreateParams,
  CustomerPaymentsListParams,
  OrderPaymentCreateParams,
};

export { default as Refund } from './data/refunds/Refund';
import { ListParameters as RefundsListParams } from './resources/refunds/parameters';
import {
  CreateParameters as PaymentRefundCreateParams,
  GetParameters as PaymentRefundGetParams,
  ListParameters as PaymentRefundsListParams,
  CancelParameters as PaymentRefundCancelParams,
} from './resources/payments/refunds/parameters';
export { RefundsListParams, PaymentRefundCreateParams, PaymentRefundGetParams, PaymentRefundsListParams, PaymentRefundCancelParams };

export { default as Shipment } from './data/orders/shipments/Shipment';
import { CreateParameters as ShipmentCreateParams, GetParameters as ShipmentGetParams, ListParameters as ShipmentsListParams, UpdateParameters as ShipmentUpdateParams } from './resources/orders/shipments/parameters';
export { ShipmentCreateParams, ShipmentGetParams, ShipmentsListParams, ShipmentUpdateParams };

export { default as Subscription } from './data/subscription/Subscription';
import {
  CreateParameters as SubscriptionCreateParams,
  GetParameters as SubscriptionGetParams,
  ListParameters as SubscriptionsListParams,
  UpdateParameters as SubscriptionUpdateParams,
  CancelParameters as SubscriptionCancelParams,
} from './resources/customers/subscriptions/parameters';
export { SubscriptionCreateParams, SubscriptionGetParams, SubscriptionsListParams, SubscriptionUpdateParams, SubscriptionCancelParams };

export { CardAudience, CardFailureReason, CardLabel, FeeRegion } from './data/global';
export { Issuer } from './data/Issuer';
export { PaymentInclude } from './data/payments/data';
