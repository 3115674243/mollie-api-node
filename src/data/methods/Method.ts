import { MethodData } from './data';
import methodHelpers from './helpers';
import Seal from '../../types/Seal';

type Method = Seal<MethodData, typeof methodHelpers>;

export default Method;

export function injectPrototypes(input: MethodData): Method {
  return Object.assign(Object.create(methodHelpers), input);
}
