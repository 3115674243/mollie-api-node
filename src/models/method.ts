import Model from '../model';

/**
 * The `Method` model
 */
export default class Method extends Model {
  image: Mollie.Image;
  constructor(props?: Partial<Mollie.MethodResponse>) {
    super(props);

    const defaults: Mollie.MethodResponse = {
      resource: 'method',
      id: null,
      description: null,
      image: {
        size1x: null,
        size2x: null,
        svg: null,
      },
      _links: {},
    };

    Object.assign(this, defaults, props);
  }

  /**
   * @param size
   * @returns {string|null}
   */
  getImage(size = '2x') {
    return this.image && (size === '1x' ? this.image.size1x : this.image.size2x);
  }
}
