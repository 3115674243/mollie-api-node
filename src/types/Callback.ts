type Callback<R> = (error: Error | null, result: R) => void;

export default Callback;
