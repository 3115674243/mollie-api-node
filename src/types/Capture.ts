declare namespace Mollie {
  interface ICapture {
    resource: string;
    id: string;
    mode: ApiMode;
    amount: IAmount;
    settlementAmount: IAmount;
    paymentId: string;
    shipmentId?: string;
    settlementId?: string;
    createdAt: string;
    _links: ILinks;
  }

  namespace Params {
    interface Create {
      // Access token parameters
      testmode?: false;
    }
  }
}
