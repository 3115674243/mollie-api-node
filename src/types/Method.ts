declare namespace Mollie {
  interface IMethod {
    resource: string;
    id: string;
    description: string;
    image: IImage;
    _links: ILinks;
  }

  namespace Method {
    enum ImageSize {
      size1x = 'size1x',
      size2x = 'size2x',
      svg = 'svg',
    }

    enum Include {
      issuers = 'issuers',
      pricing = 'pricing',
    }

    namespace Params {
      interface IGet {
        locale?: Locale;

        // Access token parameters
        profileId?: string;
        testmode?: boolean;
        include?: Array<Include>;
      }

      interface IList {
        sequenceType?: SequenceType;
        locale?: Locale;
        amount?: IAmount;
        resource?: string;
        billingCountry?: string;

        // Access token parameters
        profileId?: string;
        testmode?: boolean;
        include?: Array<Include>;
      }
    }

    namespace Callback {
      type Get = (error: any, method: IMethod) => void;
      type List = (error: any, methods: Mollie.List<IMethod>) => void;
    }
  }
}
