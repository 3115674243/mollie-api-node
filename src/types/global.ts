export type Locale =
  | 'en_US'
  | 'nl_NL'
  | 'nl_BE'
  | 'fr_FR'
  | 'fr_BE'
  | 'de_DE'
  | 'de_AT'
  | 'de_CH'
  | 'es_ES'
  | 'ca_ES'
  | 'pt_PT'
  | 'it_IT'
  | 'nb_NO'
  | 'sv_SE'
  | 'fi_FI'
  | 'da_DK'
  | 'is_IS'
  | 'hu_HU'
  | 'pl_PL'
  | 'lv_LV'
  | 'lt_LT';

export type PaymentMethod =
  | 'bancontact'
  | 'banktransfer'
  | 'belfius'
  | 'bitcoin'
  | 'creditcard'
  | 'directdebit'
  | 'eps'
  | 'giftcard'
  | 'giropay'
  | 'ideal'
  | 'inghomepay'
  | 'kbc'
  | 'paypal'
  | 'paysafecard'
  | 'sofort';

export type ApiMode = 'test' | 'live';

export interface IImage {
  size1x: string;
  size2x: string;
  svg: string;
}

export interface IUrl {
  href: string;
  type: string;
}

export interface ILinks {
  self?: IUrl;
  documentation?: IUrl;

  [link: string]: IUrl;
}

export interface IAmount {
  currency: string;
  value: string;
}

export interface IAddress {
  streetAndNumber?: string;
  postalCode?: string;
  city?: string;
  region?: string;
  country?: string;
}

export type CardLabel =
  | 'American'
  | 'Express'
  | 'Carta Si'
  | 'Carte Bleue'
  | 'Dankort'
  | 'Diners'
  | 'Club'
  | 'Discover'
  | 'JCB'
  | 'Laser'
  | 'Maestro'
  | 'Mastercard'
  | 'Unionpay'
  | 'Visa'
  | null;

export type CardFailureReason =
  | 'invalid_card_number'
  | 'invalid_cvv'
  | 'invalid_card_holder_name'
  | 'card_expired'
  | 'invalid_card_type'
  | 'refused_by_issuer'
  | 'insufficient_funds'
  | 'inactive_card';

export type CardAudience = 'consumer' | 'business' | null;

export type FeeRegion = 'intra-eu' | 'other';

export enum SequenceType {
  oneoff = 'oneoff',
  first = 'first',
  recurring = 'recurring',
}
