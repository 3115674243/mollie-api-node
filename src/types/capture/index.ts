import { ApiMode, IAmount, ILinks } from '../../types/global';

export interface ICapture {
  resource: string;
  id: string;
  mode: ApiMode;
  amount: IAmount;
  settlementAmount: IAmount;
  paymentId: string;
  shipmentId?: string;
  settlementId?: string;
  createdAt: string;
  _links: ILinks;
}
