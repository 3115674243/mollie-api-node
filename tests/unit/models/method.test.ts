import Method from '../../../src/models/Method';
import { PaymentMethod } from '../../../src/types/global';

describe('method model', () => {
  it('should instantiate with default values', () => {
    const method = new Method();

    expect(method.getImage()).toBeNull();
    expect(method.toPlainObject()).toMatchSnapshot();
  });

  it('should instantiate with given values', () => {
    const methodProps = {
      resource: 'method',
      id: PaymentMethod.ideal,
      description: 'iDEAL',
      image: {
        size1x: 'https://www.mollie.com/external/icons/payment-methods/ideal.png',
        size2x: 'https://www.mollie.com/external/icons/payment-methods/ideal%402x.png',
        svg: 'https://www.mollie.com/external/icons/payment-methods/ideal.svg',
      },
      issuers: [
        {
          resource: 'issuer',
          id: 'ideal_ABNANL2A',
          name: 'ABN AMRO',
          image: {
            size1x: 'https://www.mollie.com/external/icons/ideal-issuers/ABNANL2A.png',
            size2x: 'https://www.mollie.com/external/icons/ideal-issuers/ABNANL2A%402x.png',
            svg: 'https://www.mollie.com/external/icons/ideal-issuers/ABNANL2A.svg',
          },
        },
        {
          resource: 'issuer',
          id: 'ideal_ASNBNL21',
          name: 'ASN Bank',
          image: {
            size1x: 'https://www.mollie.com/external/icons/ideal-issuers/ASNBNL21.png',
            size2x: 'https://www.mollie.com/external/icons/ideal-issuers/ASNBNL21%402x.png',
            svg: 'https://www.mollie.com/external/icons/ideal-issuers/ASNBNL21.svg',
          },
        },
      ],
      pricing: [
        {
          description: 'The Netherlands',
          fixed: {
            value: '0.29',
            currency: 'EUR'
          },
          variable: '0',
        },
      ],
      _links: {
        self: {
          href: 'https://api.mollie.com/v2/methods/ideal',
          type: 'application/hal+json',
        },
        documentation: {
          href: 'https://docs.mollie.com/reference/v2/methods-api/get-method',
          type: 'text/html',
        },
      },
    };
    const method = new Method(methodProps);

    expect(method.getImage('1x')).toBe(methodProps.image.size1x);
    expect(method.toPlainObject()).toMatchSnapshot();
  });
});
