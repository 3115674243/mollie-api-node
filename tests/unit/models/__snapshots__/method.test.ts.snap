// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`method model should instantiate with default values 1`] = `
Object {
  "_links": Object {},
  "data": undefined,
  "description": null,
  "getImage": [Function],
  "id": null,
  "image": Object {
    "size1x": null,
    "size2x": null,
    "svg": null,
  },
  "resource": "method",
  "toPlainObject": [Function],
}
`;

exports[`method model should instantiate with given values 1`] = `
Object {
  "_links": Object {
    "self": Object {
      "href": "https://api.mollie.com/v2/methods/ideal",
      "type": "application/hal+json",
    },
  },
  "data": Object {
    "_links": Object {
      "self": Object {
        "href": "https://api.mollie.com/v2/methods/ideal",
        "type": "application/hal+json",
      },
    },
    "description": "iDEAL",
    "id": "ideal",
    "image": Object {
      "size1x": "https://www.mollie.com/images/payscreen/methods/ideal.png",
      "size2x": "https://www.mollie.com/images/payscreen/methods/ideal%402x.png",
    },
    "resource": "method",
  },
  "description": "iDEAL",
  "getImage": [Function],
  "id": "ideal",
  "image": Object {
    "size1x": "https://www.mollie.com/images/payscreen/methods/ideal.png",
    "size2x": "https://www.mollie.com/images/payscreen/methods/ideal%402x.png",
  },
  "resource": "method",
  "toPlainObject": [Function],
}
`;
